### DISCUSSION TITLE
Cannot debug Python modules via launch.vs.json

### DISCUSSION THREAD

**Discussion by vsfeedback** [Microsoft/PTVS#7676] [Enchancements]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/t/Cannot-debug-Python-modules-via-launchv/10404444)._

---
Visual Studio provides a way to debug Python programs, but fails to launch modules properly, so relative imports are failing. Consider this script (see attached project for a full example):
```python
import sys

from ..util.abc import ABC      # requires -m mymod.app.app to run

def main() -> int:
    abc = ABC("ABC")

    print(f"abc: {str(abc)} args: {sys.argv}")

    return 0

if __name__ == "__main__":
    sys.exit(main())
```
This module can be debugged via VS Code using this `launch.json`:
```
{
    "version": "0.2.0",
    "configurations": [
        {
            "name": "Python: Module",
            "type": "python",
            "request": "launch",
            "module": "mymod.app.app",
            "justMyCode": true
        }
    ]
}
```
However, Visual Studio provides no way to specify module debugging in `launch.vs.json`, so `from ..util.abc import ABC` always fails to import another module from the same package because the import uses a relative path, which requires app to have a top-level Python module.

Using `"interpreterArguments": "-m mymod.app.app"` just runs the module and fails to attach a debugger to `app.py` because all `debugpy` arguments are added after `-m`, so they are ignored.

[module-test.zip](https://aka.ms/dc/file?name=Bcfd36d3bc5f34cfeaef9ecf5c1021538638236665377900588_module-test.zip&tid=cfd36d3bc5f34cfeaef9ecf5c1021538638236665377900588)

---
#### Feedback Bot on 6/29/2023, 06:19 PM: 

(private comment, text removed)

---
### Original Solutions
(no solutions)

**Comment by StellaHuang95** [comment]
Thanks for the issue. Moving this issue to discussion as an enhancement request for comments and upvotes.

**Comment by roman-orekhov** [comment]
Still no solution for this? By the amount of comments I guess people just use VSCode instead =/

Dunno, I view this as a bug report, not an enhancement request
